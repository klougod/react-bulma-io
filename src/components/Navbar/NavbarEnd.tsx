import React, { Children, ReactElement } from 'react'
import PropTypes from 'prop-types'

export const NavbarEnd = ({ className, children }: any) => (
  <div className={`navbar-end ${className}`}>
    {Children.toArray(children).map((item: ReactElement, index) =>
      React.cloneElement(item, {
        key: index,
        className: `${item.props.className} navbar-item`
      })
    )}
  </div>
)

NavbarEnd.defaultProps = {
  className: ''
}

NavbarEnd.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
}

export default NavbarEnd
