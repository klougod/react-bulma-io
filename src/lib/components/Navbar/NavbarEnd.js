import React, { Children } from 'react'
import PropTypes from 'prop-types'

export const NavbarEnd = ({ className, children }) => (
  <div className={`navbar-end ${className}`}>
    {
      Children.toArray(children).map((item, index) => (
        React.cloneElement(item, {
          key: index,
          className: `${item.props.className} navbar-item`
        })
      ))
    }
  </div>
)

NavbarEnd.defaultProps = {
  className: ''
}

NavbarEnd.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default NavbarEnd