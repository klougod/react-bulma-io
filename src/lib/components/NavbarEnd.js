import React from 'react'
import PropTypes from 'prop-types'

export const NavbarEnd = props => {

  const {
    className,
    children
  } = props

  return (
    <div className={`navbar-end ${className}`}>
      { children.map((item, index) => {
        return React.cloneElement(item, {
          key: index,
          className: `${item.props.className} navbar-item`
        })
      })}
    </div>
  )
}

NavbarEnd.defaultProps = {
  className: ''
}

NavbarEnd.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default NavbarEnd