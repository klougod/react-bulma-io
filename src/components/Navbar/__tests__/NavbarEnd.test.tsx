import React from 'react'
import Enzyme from 'enzyme'

import NavbarEnd from '../NavbarEnd'

describe('NavbarEnd component test', function () {
  it('should render without crash without props', function () {
    const navbarEndMock = Enzyme.shallow(<NavbarEnd />)
    expect(navbarEndMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const navbarEndMockProps = {
      className: 'navbarEnd CLass',
    }
    const navbarEndMock = Enzyme.shallow(<NavbarEnd {...navbarEndMockProps} />)
    expect(navbarEndMock).toMatchSnapshot()
  })
})
