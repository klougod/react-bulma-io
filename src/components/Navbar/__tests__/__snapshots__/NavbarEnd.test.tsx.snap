// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarEnd component test should render without crash with props 1`] = `
<div
  className="navbar-end navbarEnd CLass"
/>
`;

exports[`NavbarEnd component test should render without crash without props 1`] = `
<div
  className="navbar-end "
/>
`;
